sfa_dash/static/js/probabilistic-report-handling.js

```diff
@@ -0,0 +1,1055 @@
+/*
+ *  Creates inputs for defining observation, forecast pairs for a report.
+ */
+
+var previous_constant = null;
+$(document).ready(function() {
+    function toggle_reference_dependent_metrics(){
+        /*
+         * Disables and de-selects the forecast skill metric if not all of the
+         * object pairs have reference foreasts.
+         */
+        var nulls_exist = $('.reference-forecast-value').map(function(){return $(this).val()}).get().some(x=>x=='null');
+        var skill = $('[name=metrics][value=s]'); 
+        if(nulls_exist){
+            // hide skill, insert warning
+            skill.attr('disabled', true);
+            if($('#reference-warning').length == 0){
+                $(`<span id="reference-warning" class="warning-message">
+                   (Requires reference forecast selection)</span>`
+                 ).insertAfter(skill.next());
+            }
+            if(skill.is(':checked')){
+                skill.removeAttr('checked');
+            }
+        }else{
+            // show skill remove warning
+            skill.removeAttr('disabled');
+            $('#reference-warning').remove();
+        }
+    }
+    function searchObjects(object_type, object_id){
+        /* Get a json object from the page_data object.
+         *
+         * @param {string} object_type - The type of the object to search for.
+         *     One of 'forecasts', 'sites', 'observations', 'aggregates'.
+         *
+         * @param {string} object_id - UUID of the object to search for
+         *
+         * @returns {Object} An object containing the SFA object's metadata.
+         */
+        try{
+            var objects = page_data[object_type];
+            var id_prop = object_type.slice(0, -1) + '_id';
+            var metadata = objects.find(e => e[id_prop] == object_id);
+        }catch(error){
+            return null;
+        }
+        return metadata;
+    }
+    
+
+    var variable_unit_map = {
+        'air_temperature': 'degC',
+        'wind_speed': 'm/s',
+        'ghi': 'W/m^2',
+        'dni': 'W/m^2',
+        'dhi': 'W/m^2',
+        'poa_global': 'W/m^2',
+        'relative_humidity': '%',
+        'ac_power': 'MW',
+        'dc_power': 'MW',
+        'availability': '%',
+        'curtailment': 'MW',
+    }
```

Comment:
should we put this in a shareable module?

Reply:
I think there's a bunch of common functions that could be pulled out and organized into a js module, especiallly across the different report handling files. It looks like IE doesn't support modules though.